src/EnergyPlus/WeatherManager.cc

```diff
@@ -8602,497 +8488,522 @@ namespace WeatherManager {
+                            Line.clear();
+                            break;
                         }
-                        state.dataWeatherManager->GroundTempsFCFromEPWHeader = 0.0;
-                        int actcount = 0;
-                        for (int i = 1; i <= 12; ++i) { // take the first set of ground temperatures.
-                            Pos = index(Line, ',');
-                            if (Pos != std::string::npos) {
+                        Line.erase(0, Pos + 1);
+                    }
+                    state.dataWeatherManager->GroundTempsFCFromEPWHeader = 0.0;
+                    int actcount = 0;
+                    for (int i = 1; i <= 12; ++i) { // take the first set of ground temperatures.
+                        Pos = index(Line, ',');
+                        if (Pos != std::string::npos) {
+                            state.dataWeatherManager->GroundTempsFCFromEPWHeader(i) = UtilityRoutines::ProcessNumber(Line.substr(0, Pos), errFlag);
+                            ++actcount;
+                        } else {
+                            if (len(Line) > 0) {
                                 state.dataWeatherManager->GroundTempsFCFromEPWHeader(i) =
                                     UtilityRoutines::ProcessNumber(Line.substr(0, Pos), errFlag);
                                 ++actcount;
-                            } else {
-                                if (len(Line) > 0) {
-                                    state.dataWeatherManager->GroundTempsFCFromEPWHeader(i) =
-                                        UtilityRoutines::ProcessNumber(Line.substr(0, Pos), errFlag);
-                                    ++actcount;
-                                }
-                                break;
                             }
-                            Line.erase(0, Pos + 1);
+                            break;
                         }
-                        if (actcount == 12) state.dataWeatherManager->wthFCGroundTemps = true;
+                        Line.erase(0, Pos + 1);
+                    }
+                    if (actcount == 12) state.dataWeatherManager->wthFCGroundTemps = true;
+                }
+            }
+        } break;
+        case WeatherManager::EpwHeaderType::HolidaysDST: {
+            // A1, \field LeapYear Observed
+            // \type choice
+            // \key Yes
+            // \key No
+            // \note Yes if Leap Year will be observed for this file
+            // \note No if Leap Year days (29 Feb) should be ignored in this file
+            // A2, \field Daylight Saving Start Day
+            // A3, \field Daylight Saving End Day
+            // N1, \field Number of Holidays
+            // A4, \field Holiday 1 Name
+            // A5, \field Holiday 1 Day
+            // etc.
+            // Start with Minimum number of NumHdArgs
+            uppercase(Line);
+            NumHdArgs = 4;
+            int CurCount = 0;
```

Comment:
Here's the one-line fix for the defect. When declarations were localized, this one was put too far down inside the special day loop so it was always zero.

Reply:
One line fix plus a nice little cleanup in the code.  :+1:   Thanks @mjwitte !